src/js/IAMClient.js

```diff
@@ -18,8 +20,8 @@ export function getAssumeRoleWithWebIdentityIAM(
   if (!account || !oidc || !oidc.user) return Promise.reject();
   const assumeRoleParams = {
     idToken: oidc.user.id_token,
-    RoleSessionName: 'app1',
-    roleArn: `arn:aws:iam::${account.id}:role/roleForB`,
+    roleArn: `arn:aws:iam::${account.id}:role/${roleName}`,
+    RoleSessionName: (oidc.user) ? oidc.user.profile.email : '',
```

Comment:
```suggestion
    RoleSessionName: oidc.user?.profile.email || '';
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining